### PR TITLE
TLS support

### PR DESCRIPTION
**Pull request by Kulezi**
This pull request adds basic TLS support to the driver.
- Added *tlsConfig to SessionConfig so if *tls.Config is provided in SessionConfig, all connections will use TLS.
- Updated docker-compose.yml to use a scylla instance supporting encryption
- Added some integration tests to verify if tls works, and if I created valid certificates.

For now i implemented it as minimal as possible, but maybe we should use a HostDialer interface (https://github.com/gocql/gocql/pull/1629, https://github.com/scylladb/gocql/pull/98)?
Right now the driver doesn't support Contexts, I'd add it with a PR resolving #250

### CODE REVIEW

**Review by Choraden — APPROVED** [review]

File: session_integration_test.go [hunk 1]
```diff
@@ -230,3 +233,112 @@ func TestSessionIterIntegration(t *testing.T) { // nolint:paralleltest // Integr
...
+				InsecureSkipVerify: true,
+			},
+			certPath: certPath,
+			keyPath:  keyPath,
+		},
+		{
+			name: "tls - with client verification",
+			tlsConfig: &tls.Config{
+				RootCAs:            x509.NewCertPool(),
+				InsecureSkipVerify: false,
```
Comment (avelanarius) [inline session_integration_test.go:268]: Out of curiosity: how did you make it pass correctly with client verification - yesterday it wasn't working, what was the problem?

Comment (zimnx) [reply session_integration_test.go:268]: Without setting RootCA to DB CA you shouldn't be able to verify certificate, something is fishy :)

File: testdata/config/scylla.yaml [hunk 2]
```diff
@@ -7,4 +7,5 @@ client_encryption_options:
   certificate: /etc/scylla/db.crt
   keyfile: /etc/scylla/db.key
   truststore: /etc/scylla/cadb.pem
-  require_client_auth: true
\ No newline at end of file
+  require_client_auth: true
+
```
Comment (piodul) [inline testdata/config/scylla.yaml:11]: Nit: this whitespace change should be included in the first commit.

Comment (piodul) [reply testdata/config/scylla.yaml:11]: Same goes for changes in `db.cfg` and `gen_certificates.sh`

File: session_integration_test.go (hunk 1 above)
Comment (Kulezi) [reply session_integration_test.go:268]: > Out of curiosity: how did you make it pass correctly with client verification - yesterday it wasn't working, what was the problem?

The problem was that if InsecureSkipVerify flag is false ServerName must be provided and supplying node's IP address as ServerName wouldn't work with certificates from gocql repo as they don't list any SAN's. I created new certificates with node's IP listed as a SAN, with it it works. I guess it has to do with the fact that default gocql sslOptions default to not doing client verification (setting InsecureSkipVerifyFlag to true).

**Review by piodul — APPROVED** [review]

**Pull request merged**